### DISCUSSION TITLE
How to get all these MOOSE discussions sent to my email inbox?

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#16050] [Q&A General] [answered]
I'm rather lazy because i don't spend much time reading websites such as these.  I want everything to appear in one place - my email inbox.   How can i get github to send all these discussions to my email inbox?

a

**Comment by dschwen** [comment, marked as answer]
I believe you need to "subscribe" to the repository to receive all notifications. Then I recommend you set up some email filters, unless you like to read all our issue and PR discussions as well...

**Reply by WilkAndy** [reply]
To future readers: this works, but as @dschwen says - you'll get a deluge of all PRs, issues, etc, which might be rather annoying.  Then you have to somehow filter all that stuff out and just leave the "discussions".

**Reply by WilkAndy** [reply]
As @cticenhour says below, just look at the top of the main MOOSE page, https://github.com/idaholab/moose , and you'll see a "Watch" button - click it !

**Reply by anilkunwar** [reply]
Thank you Daniel for the answer.

**Reply by hugary1995** [reply]
@WilkAndy You can actually do that with outlook as well:
![image](https://user-images.githubusercontent.com/31555204/98059358-e4b24680-1e14-11eb-87e1-b40c6237219e.png)
Depending on the app and version you are using for outlook your mileage may vary.

**Comment by WilkAndy** [comment]
OK, i will attempt that.

**Comment by dschwen** [comment]
Here is a Google AppScripts function I wrote to tag GitHub Discussions notifications with a separate label in GMail

```
function processInbox() {
  // loop over the newest message threads
  var threads = GmailApp.search("newer_than:1h");
  for (var i = 0; i < threads.length; i++) {
    // get all messages in a given thread
    var messages = threads[i].getMessages();
    for (var j = 0; j < messages.length; j++) {
      var message = messages[j];
      var body = message.getRawContent();
      // if any of the messages in the thread contains the message id tag, label the entire thread as a discussion
      if (body.indexOf("Message-ID: <idaholab/moose/repo-discussions/") > -1) {
        threads[i].addLabel(GmailApp.getUserLabelByName("GitHub/MOOSE/Discussions"));
        threads[i].removeLabel(GmailApp.getUserLabelByName("GitHub/MOOSE"));
        threads[i].moveToArchive();
        break;
      }
    }
  }
}
```

I'm running that on a timer trigger every 5mins.

Note that I automatically sort all GitHub MOOSE repo emails into the label `GitHub/MOOSE` with a regular GMail filter. The script above replaces that label with `GitHub/MOOSE/Discussions` if a certain header is detected.

**Reply by WilkAndy** [reply]
Holy Dooley!!  I have no idea how i'd do something similar with MS Outlook, which is what my organisation uses for email.   I haven't yet figured out how to subscribe to all Discussions.

**Reply by cticenhour** [reply]
To subscribe, you just need to "Watch" the repo. Look for this across from the repo title near the top right of the page. As far as filtering, I have set up automated gmail filters using the UI to label regular GitHub repo notifications and when I am mentioned, but nothing as fancy as @dschwen, where he's reading the meta-data. That's nice!

**Reply by tophmatthews** [reply]
So somewhere in this script is how I can figure out to bypass @dschwen's filters for my PR reviews? interesting...

**Reply by dschwen** [reply]
Yeah, since the filter script does not (yet) discriminate between headers and body you could try adding the string `Message-ID: <idaholab/moose/repo-discussions/` in your PR comments :-D

**Reply by lindsayad** [reply]
Oh em gee. Just watched the repo and the deluge is real. @dschwen do you have anything else that helps you discriminate github prs/issues that you're a participant in vs. all the others that you are not participating in? I feel like a turtle about to go back into his shell
